include/ignition/gui/GuiEvents.hh

```diff
@@ -201,6 +201,60 @@ namespace ignition
         /// \brief The 3D point that the user clicked within the scene.
         private: math::Vector3d point;
       };
+
+      /// \brief Event which is called to broadcast the 3D coordinates of a
+      /// user's right click within the scene.
+      class RightClickToScene : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _point The point which the user has right clicked
+        /// within the scene
+        public: explicit RightClickToScene(const math::Vector3d &_point)
+            : QEvent(kType), point(_point)
+        {
+        }
+
+        /// \brief Unique type for this event.
+        static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser - 6);
+
+        /// \brief Get the point within the scene that the user clicked.
+        /// \return The 3D point.
+        public: math::Vector3d Point() const
+        {
+          return this->point;
+        }
+
+        /// \brief The 3D point that the user clicked within the scene.
+        private: math::Vector3d point;
+      };
+
+      /// \brief Event which is called to enable or disable the right click
+      /// dropdown menu. This is primarily used by plugins which also use
+      /// the right click to cancel any actions currently in progress.
```

Comment:
So this event needs to be used together with `RightClickToScene`, right? I'm wondering if it wouldn't be enough to call it `DropdownMenu`, since it doesn't carry any information about the click.

Reply:
I just looked at how it's being used and now I think a better name may be `DropdownMenuEnabled`

Reply:
True! Updated here 1b8204e841affe2b0ca602e4e215ad557df2a020